### DISCUSSION TITLE
Behaviour change in DRF 3.15.0

### DISCUSSION THREAD

**Discussion by martinlehoux** [encode/django-rest-framework#9321] [Potential Issue] [answered]
I have a very simple test that passes with version 3.14.0, but not with version 3.15.0.

```py
def test_journey_to_suggested_station_has_correct_creation_reason(self) -> None:
        advert = AdvertFactory.create()
        closest_station = self.create_station()

        serializer = AdvertMutationSerializer(
            advert,
            data={"suggested_transportation_station": closest_station.pk},
        )
        serializer.is_valid(raise_exception=True)
        serializer.save()
```

The test fails because of a database constraint at the `serializer.save()` line, but this only shows the weird behaviour: after `serializer.is_valid(raise_exception=True)`, I have
```py
serializer.validated_data = {
	'has_filled_rules': False,
	'suggested_transportation_station': <TransportationStation: Marseilles Saint-Charles - FR>,
	'is_nearest_station_displayable': True
}
```

I don't understand why there is validated data that was not provided as input. For instance, even the `advert.has_filled_rules` value is `True` during the whole test.

- Did I miss the behaviour change that explains that?
- Or did I have a bug in my code that was fixed?

Thanks!

**Comment by martinlehoux** [comment]
I may have a hint: `has_filled_rules = models.BooleanField(default=False, blank=True)`

Maybe the default is now used as serializer default when not passed in initial data?

**Reply by lovelydinosaur** [reply, marked as answer]
Yes... gnarly unexpected side-effect of #9030.

Resolved in master... we're looking at getting a patch release out ASAP, and working on improving the maintenance plans for the future.

**Reply by martinlehoux** [reply]
Thanks a lot!
